src/main/scala/mesosphere/marathon/MarathonSchedulerService.scala

```diff
@@ -63,12 +64,17 @@ class MarathonSchedulerService @Inject() (
 
   val log = Logger.getLogger(getClass.getName)
 
-  val frameworkId = frameworkIdUtil.fetch
-  frameworkId match {
-    case Some(id) =>
-      log.info(s"Setting framework ID to ${id.getValue}")
-    case None =>
-      log.info("No previous framework ID found")
+  def frameworkId: Option[FrameworkID] = {
+    val fid = frameworkIdUtil.fetch
```

Comment:
See: https://github.com/mesosphere/mesos-utils/blob/master/src/main/scala/mesosphere/mesos/util/FrameworkIdUtil.scala#L26

The call to `org.apache.mesos.state.State.fetch()` does not pass the timeout down to the underlying Java and C++ futures.  Created #1068 in to track this and similar bugs.